src/dom/rect.js

```diff
@@ -54,11 +54,39 @@ export default class Rect {
 			enumerable: false
 		} );
 
-		if ( isElement( source ) || isRange( source ) ) {
-			source = source.getBoundingClientRect();
-		}
+		// Acquires all the rect properties from the passed source.
+		//
+		// @private
+		// @param {ClientRect|module:utils/dom/rect~Rect|Object} source}
+		const setProperties = source => {
+			rectProperties.forEach( p => this[ p ] = source[ p ] );
+		};
 
-		rectProperties.forEach( p => this[ p ] = source[ p ] );
+		if ( isElement( source ) ) {
+			setProperties( source.getBoundingClientRect() );
```

Comment:
It used to be that way. But since the new case arrived, where`width` must be set to `0` and the output of `getBoundingClientRect`/`getClientRects` is read–only, I would need to use an intermediate object to write client rect's properties to it (in a loop, `Object.assign` fails) and then alter the `witdth`, which doesn't look pretty. So I decided this looks better.